ccw.core/src/clj/ccw/eclipse.clj

```diff
@@ -620,10 +651,58 @@
         (let [s (f monitor)]
           (if (instance? IStatus s)
             s
-            (Status/OK_STATUS)))
+            Status/OK_STATUS))
         (catch Exception e
           (CCWPlugin/createErrorStatus (format "Unexpected exception while executing Job %s" name), e))))))
 
+(defn cancelable-workspace-job
+  "Create a WorkspaceJob with name, error-handler, cancel-handler and finally-handler.
+  Delegate runInWorkspace to (f monitor).
+  Takes care of returning Status/OK_STATUS, STATUS/CANCEL_STATUS or error.
+  The handlers finally-handler! and cancel-handler! should accept the
+  WorkspaceJob as parameter while error-handler should accept the
+  WorkspaceJob and the Exception (in a vector).
+
+  Note1: f should always check for (.isCanceled monitor) and break the
+  computation (throwing either InterruptedException or
+  OperationCanceledException) in order to react to user's cancel action.
```

Comment:
what's the difference between InterruptedException and OperationCanceledException?